crates/extensions/tedge_mqtt_ext/src/trie.rs

```diff
@@ -1058,12 +1102,12 @@ mod tests {
                 let mut t = MqtTrie::default();
                 for s in subscriptions {
                     let diff = t.insert(&s,1);
-                    for removed in diff.unsubscribe {
+                    assert!(diff.subscribe.len() <= 1, "Insert subscribed more than one topic at once: {diff:?}");
```

Comment:
I would add one more check, to make the contract crystal clear

```suggestion
                    assert!(diff.subscribe.len() <= 1, "Insert subscribed more than one topic at once: {diff:?}");
                    if let Some(added) = diff.subscribe.iter().next() {
                        assert_eq!(added, &s, "Insert({s}) subscribed to a different topic: {added}");
                    }

```

Reply:
Done: https://github.com/thin-edge/thin-edge.io/pull/3661/commits/1c0ca94201076d862610ca32719c3c66506f1d49